src/dune_rules/jsoo/js_of_ocaml.ml

```diff
@@ -96,30 +97,33 @@ module In_buildable = struct
         (let+ flags = Ordered_set_lang.Unexpanded.field "flags"
          and+ javascript_files =
            field "javascript_files" (repeat string) ~default:[]
-         in
+         and+ wasm_files = field "wasm_files" (repeat string) ~default:[] in
          { flags =
              { build_runtime = Ordered_set_lang.Unexpanded.standard
              ; compile = flags
              ; link =
                  flags (* we set link as well to preserve the old semantic *)
              }
          ; javascript_files
+         ; wasm_files
          })
     else
       fields
         (let+ flags = Flags.decode
          and+ javascript_files =
            field "javascript_files" (repeat string) ~default:[]
-         in
-         { flags; javascript_files })
+         and+ wasm_files = field "wasm_files" (repeat string) ~default:[] in
```

Comment:
We probably need to enable this only for newer dune lang versions. see how it's done above ` if syntax_version < (3, 0) then`